chain/chain/src/resharding/manager.rs

```diff
@@ -265,45 +229,47 @@ impl ReshardingManager {
                 retain_mode,
             )?;
 
+            // Split the parent trie and create a new child trie.
+            let trie_changes =
+                parent_trie.retain_split_shard(&split_shard_event.boundary_account, retain_mode)?;
```

Comment:
nit: Maybe just unpack all fields from the split_shard_event at the beginning of this method?